core/src/jvmMain/kotlin/com/powersync/ExtractLib.kt

```diff
@@ -26,14 +23,12 @@ internal fun extractLib(fileName: String): Path {
 
     val path = "/$prefix${fileName}_$arch.$extension"
 
-    val tmpPath = createTempFile("$prefix$fileName", ".$extension")
-    Runtime.getRuntime().addShutdownHook(Thread { tmpPath.deleteIfExists() })
+    val resourceURI =
+        (R::class.java.getResource(path) ?: error("Resource $path not found"))
 
-    (R::class.java.getResourceAsStream(path) ?: error("Resource $path not found")).use { input ->
-        tmpPath.outputStream().use { output ->
-            input.copyTo(output)
-        }
-    }
-
-    return tmpPath
+    /**
+     * Wrapping in a File handle resolves the URI to a path usable by SQLite.
+     * This is particularly relevant on Windows.
+     */
```

Comment:
> Wrapping in a File handle resolves the URI to a path usable by SQLite.

For my own understanding, is this to transform the path into the native representation (because `resourceURI.path` will always be a URL-style path which is not supported on Windows?). Maybe the comment could be a little clearer on why it's relevant.

Unrelated nit: I prefer `//` comments inside of methods, this one isn't part of any KDoc declaration either way.